msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/SignInStates.kt

```diff
@@ -382,55 +382,54 @@ class SignInPasswordRequiredState(
  * @property username: Username of the user
  * @property config Configuration used by Native Auth
  */
-abstract class SignInAfterSignUpBaseState(
+class SignInContinuationState(
     override val continuationToken: String?,
-    internal open val username: String,
+    internal val username: String,
     private val config: NativeAuthPublicClientApplicationConfiguration
 ) : BaseState(continuationToken), State, Serializable {
-    private val TAG: String = SignInAfterSignUpBaseState::class.java.simpleName
+    private val TAG: String = SignInContinuationState::class.java.simpleName
 
     /**
-     * SignInAfterSignUpCallback receives the result for sign in after signup for Native Auth
+     * SignInContinuationCallback receives the result for sign in after flow completion for Native Auth
      */
-    interface SignInAfterSignUpCallback : Callback<SignInResult>
+    interface SignInContinuationCallback : Callback<SignInResult>
 
     /**
-     * Submits the sign-in-after-sign-up verification code to the server; callback variant.
+     * Submits the sign-in-continuation verification code to the server; callback variant.
      *
      * @param scopes (Optional) The scopes to request.
-     * @param callback [com.microsoft.identity.nativeauth.statemachine.states.SignInAfterSignUpBaseState.SignInAfterSignUpCallback] to receive the result on.
-     * @return The results of the sign-in-after-sign-up action.
+     * @param callback [com.microsoft.identity.nativeauth.statemachine.states.SignInContinuationState.SignInContinuationCallback] to receive the result on.
+     * @return The results of the sign-in-continuation action.
      */
-    fun signInAfterSignUp(scopes: List<String>? = null, callback: SignInAfterSignUpCallback) {
-        LogSession.logMethodCall(TAG, "${TAG}.signInAfterSignUp")
+    fun signIn(scopes: List<String>? = null, callback: SignInContinuationCallback) {
+        LogSession.logMethodCall(TAG, "${TAG}.signIn")
 
         NativeAuthPublicClientApplication.pcaScope.launch {
             try {
-                val result = signInAfterSignUp(scopes)
+                val result = signIn(scopes)
                 callback.onResult(result)
             } catch (e: MsalException) {
-                Logger.error(TAG, "Exception thrown in signInAfterSignUp", e)
+                Logger.error(TAG, "Exception thrown in signIn", e)
                 callback.onError(e)
             }
         }
     }
 
     /**
-     * Submits the sign-in-after-sign-up verification code to the server; Kotlin coroutines variant.
+     * Submits the sign-in-continuation verification code to the server; Kotlin coroutines variant.
      *
      * @param scopes (Optional) The scopes to request.
      * @return The results of the sign-in-after-sign-up action.
      */
-    suspend fun signInAfterSignUp(scopes: List<String>? = null): SignInResult {
+    suspend fun signIn(scopes: List<String>? = null): SignInResult {
```

Comment:
For those SignInCommandResult.CodeRequired & SignInCommandResult.PasswordRequired & INativeAuthCommandResult.Redirect, should we group them togerther into the error? @GBakolas is proposing a SignInCompletionError class with no utility methods. I'm guessing this is another PR after this one.

Reply:
Not sure what you mean by "group them together into the error" - which error are you referring to? Each error type will return a different state, so I don't see how we can combine them.

Reply:
When signing in with a continuation token, I don't think the user is required to provide a password / code. My initial thought was if we could do something here instead of having the sample app treat them as invalid states. But you're right, the results come with different state/error parameters and we can't combine them.